lib/EnsEMBL/REST/Model/ga4gh/Beacon.pm

```diff
@@ -541,61 +553,77 @@ sub variant_exists {
         next;
     }
 
-    # Precise match for snv or small indels - end does not make a difference 
-    if ($sv == 0 && ($seq_region_start != $new_start) && ($seq_region_end != $new_end)) {
+    # Precise match for snv or small indels - end does not make a difference
+    if ($sv == 0 && ($seq_region_start != $start_pos) && ($seq_region_end != $end_pos)) {
       next;
     }
     # Match for structural variants
     if($sv == 1){
-      next unless ($seq_region_start >= $new_start && $seq_region_start <= $start_max_pos && $seq_region_end >= $end_min_pos && $seq_region_end <= $new_end);
-    }
-
-    # All datasets where variant was found
-    my $datasets_found = $vf->get_all_VariationSets();
-    foreach my $set (@$datasets_found) {
-      $dataset_var_found{$set->dbID()} = $set;
+      next unless ($seq_region_start >= $start_pos && $seq_region_start <= $start_max_pos && $seq_region_end >= $end_min_pos && $seq_region_end <= $end_pos);
     }
 
     # Variant is a SNV
     if ($sv == 0) {
     $ref_allele_string = $vf->ref_allele_string();
     $alt_alleles = $vf->alt_alleles();
 
-    if ((uc($ref_allele_string) eq uc($new_ref))
-      && (grep(/^$new_alt$/i, @{$alt_alleles}))) {
+    # Some ins/del have allele string 'G/-', others have TG/T
+    # trim sequence doesn't always work
+    if ((uc($ref_allele_string) eq uc($ref_allele) || (uc($ref_allele_string) eq uc($new_ref)))
+      && (grep(/^$alt_allele$/i, @{$alt_alleles}) || grep(/^$new_alt$/i, @{$alt_alleles}))) {
```

Comment:
please use the block form of grep, i.e `grep { /^$alt_allele$/i } @{$alt_alleles}`
perhaps also update the string boundaries to \A \z instead of ^ $.

Reply:
`contains_value` used instead.